### DISCUSSION TITLE
How does OC update Document.Id type?

### DISCUSSION THREAD

**Discussion by ShaneCourtrille** [OrchardCMS/OrchardCore#15168] [General]
@MikeAlhayek As part of #13997 the document Id type was changed from int to long.  I'm curious in which process OC uses to upgrade the actual database?  

We have a customized tenant provisioning process and it looks like we need to do something similar as we're seeing SQL errors like "Microsoft.Data.SqlClient.SqlException (0x80131904): Column 'Document.Id' is not the same data type as referencing column 'ContentItemIndex.DocumentId' in foreign key 'FK_ContentItemIndex'. Could not create constraint or index." when we try to run the recipe step.  I think one thing that makes our use case a bit weird is that we delete tenants without deleting the actual database and then just reuse it again later which has worked fine until now.

Thanks

**Comment by ShaneCourtrille** [comment]
I just realized that ContentItemIndex might be a custom table we have created which would explain why it has a problem.  That said I'm still curious as to the process since I don't see any update happening to our existing tenants.

**Reply by MikeAlhayek** [reply]
@ShaneCourtrille 

ContentItemIndex is OC table. If you setup new site, all tables will use long by default. However, for existing sites, integer should be used in all the tables. Unless you have a crisis, I would say there is no reason to upgrade your existing tenants from int to long.
